### Get familiar with provided data


In [1]:
import pandas as pd

#### Find the Metro Mont Royal and Berri stations codes

In [3]:
stations_names = pd.read_csv('../Downloaded_data/2014/Stations_2014.csv')

In [15]:
stations_names[stations_names['name'].str.contains('Mont')]


code                                               name   latitude  \
7    6149                           Chapleau / du Mont-Royal  45.538670   
18   6223                            du Mont-Royal / du Parc  45.517000   
30   6380  Parc J.-Arthur-Champagne (Chambly / du Mont-Ro...  45.551584   
37   6221                              du Mont-Royal / Clark  45.519410   
50   6157                        de la Roche / du Mont-Royal  45.528746   
55   6184          Métro Mont-Royal (Rivard / du Mont-Royal)  45.524673   
58   6169                              Boyer / du Mont-Royal  45.527426   
67   6152                             Chabot / du Mont-Royal  45.534134   
73   6057            Peel / avenue des Canadiens de Montréal  45.497697   
76   6902                                 Montcalm / Ontario  45.521094   
79   6154                          Marquette / du Mont-Royal  45.532290   
90   6183                              Laval / du Mont-Royal  45.522413   
94   6182                         de Bullion / du Mont-Royal  45.521500   
104  6155                            Garnier / du Mont-Royal  45.530920   
107  6087                        Notre-Dame / de la Montagne  45.493029   
160  6065                        de la Montagne / Sherbrooke  45.499677   
182  6229                          Coloniale / du Mont-Royal  45.521156   
229  5006                          Collège Édouard-Montpetit  45.537226   
238  6112                          Montcalm / de Maisonneuve  45.519157   
287  6306                     du Mont-Royal / Vincent-d'Indy  45.510086   
300  6927                       Édouard-Montpetit / Stirling  45.507230   
356  6727                        de Montmorency / Richardson  45.484896   
381  6151                        des Érables / du Mont-Royal  45.535650   
392  6187                            Resther / du Mont-Royal  45.525643   
398  6150                            Messier / du Mont-Royal  45.537114   
424  6222                       Jeanne Mance / du Mont-Royal  45.517643   
428  6383                       Bourbonnière / du Mont-Royal  45.553571   
434  6502                                 Casino de Montréal  45.506373   

     longitude  
7   -73.569360  
18  -73.589000  
30  -73.561916  
37  -73.586850  
50  -73.578654  
55  -73.582550  
58  -73.579918  
67  -73.573492  
73  -73.568646  
76  -73.562251  
79  -73.575440  
90  -73.584130  
94  -73.585100  
104 -73.576740  
107 -73.564818  
160 -73.578849  
182 -73.585589  
229 -73.495067  
238 -73.557860  
287 -73.611429  
300 -73.615085  
356 -73.560119  
381 -73.571720  
392 -73.581748  
398 -73.571003  
424 -73.588928  
428 -73.560494  
434 -73.524577

In [16]:
stations_names[stations_names['name'].str.contains('Métro Mont-Royal')]

code                                       name   latitude  longitude
55  6184  Métro Mont-Royal (Rivard / du Mont-Royal)  45.524673  -73.58255

In [17]:
stations_names[stations_names['name'].str.contains('Berri')]

code                         name   latitude  longitude
70   6019           Berri / Sherbrooke  45.517910 -73.567143
105  6015       Berri / de Maisonneuve  45.515299 -73.561273
130  6173             Berri / Cherrier  45.519146 -73.569241
262  6210  Métro Sauvé (Sauvé / Berri)  45.550692 -73.656367
362  6023        de la Commune / Berri  45.510860 -73.549830
429  6192          Berri / St-Grégoire  45.529802 -73.591766

Berri station has location 45.5153, -73.5613 (googled). The exact match with Berri/de Maisonneuve

#### Needed codes 6184 ---> 6015

#### Explore rides

In [11]:
rides = pd.read_csv('../Downloaded_data/2014/OD_2014-04.csv')

In [18]:
rides.head(2)

start_date  start_station_code          end_date  end_station_code  \
0  2014-04-15 00:01                6209  2014-04-15 00:18              6436   
1  2014-04-15 00:01                6214  2014-04-15 00:11              6248   

   duration_sec  is_member  
0          1061          1  
1           615          1

check if the rest of the data the same

In [21]:
stations_names_2015 = pd.read_csv('../Downloaded_data/2015/Stations_2015.csv')

In [22]:
stations_names_2015[stations_names_2015['name'].str.contains('Mont')]

code                                               name   latitude  \
10   6155                            Garnier / du Mont-Royal  45.530920   
19   6221                              du Mont-Royal / Clark  45.519410   
34   6223                            du Mont-Royal / du Parc  45.517000   
46   6182                         de Bullion / du Mont-Royal  45.521500   
52   6154                          Marquette / du Mont-Royal  45.532290   
54   6157                        de la Roche / du Mont-Royal  45.528746   
73   6151                        des Érables / du Mont-Royal  45.535650   
98   6152                             Chabot / du Mont-Royal  45.534134   
99   6169                              Boyer / du Mont-Royal  45.527426   
118  6927                       Édouard-Montpetit / Stirling  45.507230   
125  6902                                 Montcalm / Ontario  45.521094   
138  6065                        de la Montagne / Sherbrooke  45.499677   
144  6229                          Coloniale / du Mont-Royal  45.521156   
147  6187                            Resther / du Mont-Royal  45.525643   
159  6112                          Montcalm / de Maisonneuve  45.519157   
160  6057            Peel / avenue des Canadiens de Montréal  45.497697   
172  6183                              Laval / du Mont-Royal  45.522413   
183  6222                       Jeanne Mance / du Mont-Royal  45.517643   
207  6184          Métro Mont-Royal (Rivard / du Mont-Royal)  45.524673   
216  6149                           Chapleau / du Mont-Royal  45.538670   
231  6087                        Notre-Dame / de la Montagne  45.493029   
252  6150                            Messier / du Mont-Royal  45.537114   
290  6727                        de Montmorency / Richardson  45.484896   
310  6306                     du Mont-Royal / Vincent-d'Indy  45.510086   
340  6380  Parc J.-Arthur-Champagne (Chambly / du Mont-Ro...  45.551584   
342  5006                          Collège Édouard-Montpetit  45.537226   
413  6383                       Bourbonnière / du Mont-Royal  45.553571   
444  6502                                 Casino de Montréal  45.506373   

     longitude  
10  -73.576740  
19  -73.586850  
34  -73.589000  
46  -73.585100  
52  -73.575440  
54  -73.578654  
73  -73.571720  
98  -73.573492  
99  -73.579918  
118 -73.615085  
125 -73.562251  
138 -73.578849  
144 -73.585589  
147 -73.581748  
159 -73.557860  
160 -73.568646  
172 -73.584130  
183 -73.588928  
207 -73.582550  
216 -73.569360  
231 -73.564818  
252 -73.571003  
290 -73.560119  
310 -73.611429  
340 -73.561916  
342 -73.495067  
413 -73.560494  
444 -73.524577

In [23]:
stations_names_2015[stations_names_2015['name'].str.contains('Berri')]

code                         name   latitude  longitude
2    6192          Berri / St-Grégoire  45.529802 -73.591766
171  6210  Métro Sauvé (Sauvé / Berri)  45.550692 -73.656367
181  6023        de la Commune / Berri  45.510860 -73.549830
212  6015       Berri / de Maisonneuve  45.515299 -73.561273
367  6173             Berri / Cherrier  45.519146 -73.569241
441  6019           Berri / Sherbrooke  45.517910 -73.567143

all the data has the same structure. 